packages/accounts-base/accounts_client_tests.js

```diff
@@ -304,3 +304,26 @@ Tinytest.addAsync(
   }
 );
 
+Tinytest.addAsync('accounts - Session storage', async (test, done) => {
+
+  Accounts.config({ useSessionStorage: true })
+  test.isTrue(Accounts._options.useSessionStorage)
+
+  // Login a user with AccountClientSession and test that tokens are in sessionStorage
+  logoutAndCreateUser(test, done, () => {
+    test.isNotUndefined(sessionStorage.getItem('Meteor.loginToken'))
+    test.isUndefined(localStorage.getItem('Meteor.loginToken'))
+    Accounts.logout()
+    removeTestUser()
+  })
+
+  // Login a user with AccountClientStorage and test that tokens are in localStorage
+  logoutAndCreateUser(test, done, () => {
+    Accounts.config({ useSessionStorage: false })
+    test.isFalse(Accounts._options.useSessionStorage)
+    test.isUndefined(sessionStorage.getItem('Meteor.loginToken'))
+    test.isNotUndefined(localStorage.getItem('Meteor.loginToken'))
+    Accounts.logout()
+    removeTestUser()
+  })
+})
```

Comment:
Here are some suggested improvements:

- The `done` callback in the test definition controls test finalization. Using `done` twice per test might cause only one to evaluate successfully during the async process.
  - Possible fixes:
    - Nest them, ensuring verification of session storage occurs after local storage.
    - Utilize `testAsyncMulti` and create two test cases: one for `useSessionStorage: true` and the other for `useSessionStorage: false` or default behavior. This approach appears cleaner. See example: https://github.com/meteor/meteor/blob/release-2.16/packages/accounts-password/email_tests.js#L11-L70 (be aware of expect usage in the example)
- Consider renaming the test to something inclusive of both storage types, such as "Accounts - Local and Session Storage."
- While `useSessionStorage` seems appropriate, note that the `useXX` convention is widely adopted in React for hooks. However, this likely isn't a significant issue. I'm fine with it.

Reply:
For the `useSessionStorage`, the alternative I can think of would be to something like `storage: 'session'` and `storage: 'local'`.

Reply:
I prefer those actually. Anyway, can we also act on the types for autocompletion? `accounts-base.d.ts` should be.

Reply:
Done and in the end using `clientStorage` for more flexibility.

Reply:
Something is missing, the tests fail.

![image](https://github.com/meteor/meteor/assets/2581993/4480a14b-e414-4c62-91ca-2a9fa466d3cd)

Reply:
Fixed the issue, but have another in the tests. Either I somehow need to reset the server with new config or maybe change the order to have the default first (and not setting it via options).

Reply:
https://github.com/meteor/meteor/blob/b209f24a7896dc72b186003f199b2c100ae159f7/packages/accounts-base/accounts_common.js#L281

Maybe we can disable this check on tests.

Reply:
If that is fine with you. I'm going to try this in the next commit.

Reply:
Or I can make even more strict restriction:
```javascript
if (key !== 'collection' && (Meteor.isTest && key !== 'clientStorage')) {
            throw new Meteor.Error(`Can't set \`${key}\` more than once`);
          }
```

Reply:
That did not seem to work :(